src/Lut.ts

```diff
@@ -475,84 +478,6 @@ export function remapControlPoints(
-
-    if (cp.x < 0 && cpNext.x > 0) {
-      // interpolate
-      const pct = (0 - cp.x) / (cpNext.x - cp.x);
-      cp.opacity = lerp(cp.opacity, cpNext.opacity, pct);
-      cp.color[0] = lerp(cp.color[0], cpNext.color[0], pct);
-      cp.color[1] = lerp(cp.color[1], cpNext.color[1], pct);
-      cp.color[2] = lerp(cp.color[2], cpNext.color[2], pct);
-      // shift cp to 0
-      cp.x = 0;
-    } else if (cp.x > 255 && cpPrev.x < 255) {
-      // interpolate
-      const pct = (cp.x - 255) / (cp.x - cpPrev.x);
-      cp.opacity = lerp(cpPrev.opacity, cp.opacity, pct);
-      cp.color[0] = lerp(cpPrev.color[0], cp.color[0], pct);
-      cp.color[1] = lerp(cpPrev.color[1], cp.color[1], pct);
-      cp.color[2] = lerp(cpPrev.color[2], cp.color[2], pct);
-      // shift cp to 255
-      cp.x = 255;
-    }
-    if (cp.x >= 0 && cp.x <= 255) {
-      resultControlPoints.push(cp);
-    }
-  }
 
-  // lastly, add a point for start and end if needed.
-  if (resultControlPoints[0].x !== 0) {
-    // if the first 2 points have same opacity and color, then just shift first pt to 0.
-    if (
-      resultControlPoints.length > 1 &&
-      resultControlPoints[0].opacity === resultControlPoints[1].opacity &&
-      resultControlPoints[0].color[0] === resultControlPoints[1].color[0] &&
-      resultControlPoints[0].color[1] === resultControlPoints[1].color[1] &&
-      resultControlPoints[0].color[2] === resultControlPoints[1].color[2]
-    ) {
-      resultControlPoints[0].x = 0;
-    } else {
-      // otherwise, add a point at 0.
-      resultControlPoints.unshift({
-        x: 0,
-        opacity: resultControlPoints[0].opacity,
-        color: resultControlPoints[0].color,
-      });
-    }
-  }
-  if (resultControlPoints[resultControlPoints.length - 1].x !== 255) {
-    // if the last 2 points have same opacity and color, then just shift last pt to 255.
-    if (
-      resultControlPoints.length > 1 &&
-      resultControlPoints[resultControlPoints.length - 1].opacity ===
-        resultControlPoints[resultControlPoints.length - 2].opacity &&
-      resultControlPoints[resultControlPoints.length - 1].color[0] ===
-        resultControlPoints[resultControlPoints.length - 2].color[0] &&
-      resultControlPoints[resultControlPoints.length - 1].color[1] ===
-        resultControlPoints[resultControlPoints.length - 2].color[1] &&
-      resultControlPoints[resultControlPoints.length - 1].color[2] ===
-        resultControlPoints[resultControlPoints.length - 2].color[2]
-    ) {
-      resultControlPoints[resultControlPoints.length - 1].x = 255;
-    } else {
-      // otherwise, add a point at 255.
-      resultControlPoints.push({
-        x: 255,
-        opacity: resultControlPoints[resultControlPoints.length - 1].opacity,
-        color: resultControlPoints[resultControlPoints.length - 1].color,
-      });
-    }
-  }
-  return resultControlPoints;
+  return newControlPoints;
```

Comment:
Removing this big chunk of range-handling logic and returning `newControlPoints` instead looks very clean!